vhdbuilder/packer/test-scan-and-cleanup.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+retrycmd_if_failure() {
+  RETRIES=${1}; WAIT_SLEEP=${2}; CMD=${3}; TARGET=$(basename ${3} .sh)
+  echo "##[group]$TARGET" >> ${TARGET}-output.txt
+  echo -e "Running ${CMD} with ${RETRIES} retries" >> ${TARGET}-output.txt
+  for i in $(seq 1 ${RETRIES}); do
+    ${CMD} >> ${TARGET}-output.txt 2>&1 && break ||
+    if [ ${i} -eq ${RETRIES} ]; then
+      sed -i "3i ${TARGET} failed ${i} times" ${TARGET}-output.txt
+      echo "##[endgroup]$TARGET" >> ${TARGET}-output.txt
+      cat ${TARGET}-output.txt && rm ${TARGET}-output.txt
+      exit 1
+    else
+      sleep ${WAIT_SLEEP}
+      echo -e "\n\nNext Attempt:\n\n" >> ${TARGET}-output.txt
+    fi
+  done
+  echo "##[endgroup]$TARGET" >> ${TARGET}-output.txt
+  cat ${TARGET}-output.txt && rm ${TARGET}-output.txt
+}
+
+if [[ -z "$SIG_GALLERY_NAME" ]]; then
+  SIG_GALLERY_NAME="PackerSigGalleryEastUS"
+fi
+
+SCRIPT_ARRAY+=("./vhdbuilder/packer/cleanup.sh") # Always run cleanup
+
+# Check to ensure the build step succeeded
+SIG_VERSION=$(az sig image-version show \
+-e ${CAPTURED_SIG_VERSION} \
+-i ${SIG_IMAGE_NAME} \
+-r ${SIG_GALLERY_NAME} \
+-g ${AZURE_RESOURCE_GROUP_NAME} \
+--query id --output tsv || true)
```

Comment:
we should generally discourage `|| true` imo, would prefer to see failures as they occur rather than silencing them and relying on fall back logic

Reply:
Thanks. Artifact from test/scan scripts that have set -e declared. If the build failed, this command caused the test/scan/cleanup to fail if I didn't || true. Will change because should be fine without set -e.